File: cibcc/Go/step1/cibcc.go
package main

import (
	"os"
	"fmt"
)

func main() {
	if len(os.Args) != 2 {
		fmt.Println("引数の個数が正しくありません")
	}

	input := os.Args[1]

	fmt.Println(".intel_syntax noprefix")
	fmt.Println(".global main")
	fmt.Println("main:")

	fmt.Println("  mov rax, ", input)

	fmt.Println("  ret")
}

File: cibcc/python/step1/cibcc.py
import sys

args = sys.argv
if len(args) != 2:
	print("引数の個数が正しくありません")
	sys.exit()

input = sys.argv[1]

print(".intel_syntax noprefix")
print(".global main");
print("main:")

print("  mov rax, ", input)

print("  ret")

